### PR TITLE
reorganizing the cli to a more idiomatic organization

### PR DESCRIPTION
**Pull request by edaniszewski**
@timfallmk  -- let me know what you think of this structure. There is still work that can be done to clean up / consolidate stuff, but just wanted to get feedback on the structure so far, as well as any other suggestions for it. 

I still need to go through and update CI/docs/stuff like that for this. Will work on that next.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +0,0 @@
-FROM golang
```
Comment (timfallmk) [inline Dockerfile]: Maybe we need a tracking issue to get this back in at some point.

**Comment by timfallmk** [comment]
Fixes #105

File: Dockerfile (hunk 1 above)
Comment (timfallmk) [reply Dockerfile]: Added #106

Comment (edaniszewski) [reply Dockerfile]: thanks!

**Comment by edaniszewski** [comment]
@timfallmk I nested `cmd/synse/synse.go` because I'm just a sheep and was following a guide that said to do it that way. I've also seen other project (e.g. docker-cli) that do it that way, so I figured it was the 'right' way.

I don't currently envision anything else in there, so it makes more sense to not nest, I can update.

**Pull request merged**